ocp_resources/resource.py

```diff
@@ -807,17 +807,17 @@ def delete(self, wait: bool = False, timeout: int = TIMEOUT_4MINUTES, body: Dict
         self.logger.info(f"Delete {self.kind} {self.name}")
 
         if self.exists:
-            hashed_data = self.hash_resource_dict(resource_dict=self.instance.to_dict())
-            self.logger.info(f"Deleting {hashed_data}")
-            self.logger.debug(f"\n{yaml.dump(hashed_data)}")
+            try:
+                hashed_data = self.hash_resource_dict(resource_dict=self.instance.to_dict())
+                self.logger.info(f"Deleting {hashed_data}")
+                self.logger.debug(f"\n{yaml.dump(hashed_data)}")
 
-        try:
-            self.api.delete(name=self.name, namespace=self.namespace, body=body)
-            if wait:
-                self.client_wait_deleted(timeout=timeout)
+                self.api.delete(name=self.name, namespace=self.namespace, body=body)
```

Comment:
If someone calls delete, we expect the resource to exist and not be automatically deleted by the product design. This change would hide such problems. Is that intentional?